dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/ProcessingManager.java

```diff
@@ -85,6 +85,9 @@ private ProcessingManager(Configuration config) throws InvalidConfigurationExcep
         final FunctionProcessor removeDeadJumps = RemoveDeadCondJumps.fromConfig(config);
         programProcessors.addAll(Arrays.asList(
                 printBeforeProcessing ? DebugPrint.withHeader("Before processing", Printer.Mode.ALL) : null,
+                ProgramProcessor.fromFunctionProcessor(
+                    NormalizeLoops.newInstance(), Target.FUNCTIONS, true
+                ),
```

Comment:
This will not apply to litmus-style code in case you also intend to run it on such code (e.g., SPIRV).
IIRC, back in the days the distinction between functions and threads was made to just avoid running passes on litmus code altogether. But I'm not sure if this is still necessary.
The cleanest approach would probably be to construct a separate processing pipeline for litmus code anyways.

Reply:
I hardly believe anybody will ever write litmus benchmarks that would require this pass to be applied (the same probably applies to other passes we have).

The new code follows the same pattern we use for the whole class: if the pass is a `FunctionProcessor`, then use `Target.FUNCTIONS` until threads are created. After that, we use `Target.THREADS`.

That being said, you are right that eventually we might want to have different processing pipelines for the different formats.